compose/service.py

```diff
@@ -951,9 +952,26 @@ def get_container_data_volumes(container, volumes_option):
 
         # Copy existing volume from old container
         volume = volume._replace(external=volume_path)
-        volumes.append(build_volume_binding(volume))
-
-    return dict(volumes)
+        volumes.append(volume)
+
+    return volumes
+
+
+def warn_on_masked_volume(volumes_option, container_volumes, service):
+    container_volumes = dict(
+        (volume.internal, volume.external)
+        for volume in container_volumes)
+
+    for volume in volumes_option:
+        if container_volumes.get(volume.internal) != volume.external:
+            log.warn((
+                "Service \"{service}\" is using volume \"{volume}\" from the "
+                "previous container. Host mapping \"{host_path}\" has no effect. "
+                "Remove the existing containers to use the host volume mapping."
```

Comment:
Ya, I started with that, but it gets a bit messy if they've used a different name for the command. I would use `sys.argv[0]`, but that often has a full path on it, so I'd have to `dirname()` to truncate.

I could make it `"%s rm %s" % (dirname(sys.argv[0]), service)`